src/solvers/dgsem_tree/indicators_2d.jl

```diff
@@ -190,6 +190,138 @@ function (löhner::IndicatorLöhner)(u::AbstractArray{<:Any,4},
 end
 
 
+# this method is used when the indicator is constructed as for shock-capturing volume integrals
+function create_cache(::Type{IndicatorIDP}, equations::AbstractEquations{2}, basis::LobattoLegendreBasis)
+
+  A = Array{real(basis), ndims(equations)}
+  indicator_threaded = [A(undef, nnodes(basis), nnodes(basis)) for _ in 1:Threads.nthreads()]
+
+  var_max_threaded = [A(undef, nnodes(basis), nnodes(basis)) for _ in 1:Threads.nthreads()]
+  var_min_threaded = [A(undef, nnodes(basis), nnodes(basis)) for _ in 1:Threads.nthreads()]
+
+  P_plus_threaded = [A(undef, nnodes(basis), nnodes(basis)) for _ in 1:Threads.nthreads()]
+  P_minus_threaded = [A(undef, nnodes(basis), nnodes(basis)) for _ in 1:Threads.nthreads()]
+
+  alpha_plus_threaded = [A(undef, nnodes(basis), nnodes(basis)) for _ in 1:Threads.nthreads()]
+  alpha_minus_threaded = [A(undef, nnodes(basis), nnodes(basis)) for _ in 1:Threads.nthreads()]
+
+  alpha1_threaded = [A(undef, nnodes(basis)+1, nnodes(basis), ) for _ in 1:Threads.nthreads()]
+  alpha2_threaded = [A(undef, nnodes(basis), nnodes(basis)+1, ) for _ in 1:Threads.nthreads()]
+
+  alpha_prov = A(undef, nnodes(basis), nnodes(basis), ) # TODO: Save for all elements
```

Comment:
Since we will eventually use it for AMR, we should follow the approach we use for other containers (allocate something and `resize!` it when necessary, map the buffer to a multidimensional array for convenience). See for example 
https://github.com/trixi-framework/Trixi.jl/blob/53406c4d8a26e835d96c7e2fe575d51ea2f3f9be/src/solvers/dgsem_tree/containers_1d.jl#L9-L17 
and 
https://github.com/trixi-framework/Trixi.jl/blob/53406c4d8a26e835d96c7e2fe575d51ea2f3f9be/src/solvers/dgsem_tree/containers_1d.jl#L23-L47